File: Part 2 - Regression/Section 8 - Decision Tree Regression/decision_tree_regression.R
# Decision Tree Regression
setwd("~/work/mlatoz/Part 2 - Regression/Section 8 - Decision Tree Regression")
# install.packages('rpart')
library(rpart)

# Importing the dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]


# Decision tree regressor
regressor = rpart(formula = Salary ~ ., data = dataset, control = rpart.control(minsplit = 1))

# ====> graph
level_grid = seq(min(dataset$Level), max(dataset$Level), by=0.01)
svr_plot = ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') + 
  geom_line(aes(x = level_grid, y = predict(regressor, newdata = data.frame(Level =  level_grid))), color = 'blue') +
  ggtitle('Level vs Salary (Decision Tree regression)') +
  xlab('Level') + 
  ylab('Salary')

ggsave(filename = 'r_dtr_regression.png', svr_plot)

File: Part 2 - Regression/Section 8 - Decision Tree Regression/decision_tree_regression.py
# Decision Tree Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.tree import DecisionTreeClassifier

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
level = dataset.iloc[:, 1:2].values
salary = dataset.iloc[:, 2].values


# ====> Decision Tree Regression
regressor = DecisionTreeClassifier(random_state = 0)
regressor.fit(level, salary)

# ====> Predicting a new result
salary_predict = regressor.predict(6.5)


# ====> Visualising the decision tree results
level_grid = np.arange(min(level), max(level), step = 0.01)
level_grid = level_grid.reshape(len(level_grid), 1)
plt.scatter(level, salary, color='red')
plt.plot(level_grid, regressor.predict(level_grid), color = 'blue')
plt.title('Level vs Salary (Decision Tree Regression)')
plt.xlabel('Level')
plt.ylabel('Salary')
plt.savefig('py_decision_tree.png')
plt.close()